lib/hanami/model/association.rb

```diff
@@ -0,0 +1,23 @@
+require 'hanami/model/associations/has_many'
+
+module Hanami
+  module Model
+    # Association factory
+    #
+    # @since x.x.x
+    # @api private
+    class Association
+      # Instantiate an association
+      #
+      # @since x.x.x
+      # @api private
+      def self.new(repository, target, subject)
+        case repository.root.associations[target]
+        when ROM::SQL::Association::OneToMany then Associations::HasMany
+        else
+          raise 'unsupported association'
+        end.new(repository, repository.root.name.to_sym, target, subject)
```

Comment:
@AMHOL This looks better, but I would prefer a _fail fast_ strategy. I'd move `association_class.new` into `#initialize`, so it would blow instantly when try to instantiate an unknown type of association. 😉